packages/editor/src/store/utils/notice-builder.js

```diff
@@ -95,13 +95,17 @@ export function getNotificationArgumentsForSaveFail( data ) {
 	// If the post was being published, we show the corresponding publish error message
 	// Unless we publish an "updating failed" message
 	const messages = {
-		publish: __( 'Publishing failed' ),
-		private: __( 'Publishing failed' ),
-		future: __( 'Scheduling failed' ),
+		publish: __( 'Publishing failed.' ),
+		private: __( 'Publishing failed.' ),
+		future: __( 'Scheduling failed.' ),
 	};
-	const noticeMessage = ! isPublished && publishStatus.indexOf( edits.status ) !== -1 ?
+	let noticeMessage = ! isPublished && publishStatus.indexOf( edits.status ) !== -1 ?
 		messages[ edits.status ] :
-		__( 'Updating failed' );
+		__( 'Updating failed.' );
+
+	if ( ! ( /<\/?[^>]*>/.test( error.message ) ) ) {
+		noticeMessage = noticeMessage + ' ' + error.message;
```

Comment:
The concatenation† here has me thinking about localization: Based on the examples you shared at https://github.com/WordPress/gutenberg/issues/14644#issuecomment-489077887, it seems we can generally expect that the messages are run through `__` on the server, but it's worth checking that the user's locale preferences are in-fact represented in the UI (that the user context is known for these requests, the locale data is loaded for the user, the messages themselves are localized).

† Concatenation _can be_ [discouraged](https://codex.wordpress.org/I18n_for_WordPress_Developers#Best_Practices), though typically more problematic in cases of fragments within a sentence, rather than as two separate sentences.

Reply:
It really depends when the requests dies, to if the users language is selected. If for example, there is a PHP fatal error or database error, it will load [wp_load_translations_early](https://developer.wordpress.org/reference/functions/wp_load_translations_early/), which will it's best to work users language but doesn't work 100% of the time. 

Maybe it is better to just use a mixture of the error code and the error status and keep the translations in javascript. This may also make gutenberg a little more platform agnostic.

Reply:
> Maybe it is better to just use a mixture of the error code and the error status and keep the translations in javascript. This may also make gutenberg a little more platform agnostic.

Combined with the [above discussion](https://github.com/WordPress/gutenberg/pull/15657/files#r285184131) about how not all messages can be used directly from the response, this sounds promising. I'm not as sure that it benefits platform agnosticism, if the error codes are quite specific to WordPress (moreso than a `message`, which could be any text).

Reply:
@aduth Updated PR to use sprintf https://github.com/WordPress/gutenberg/pull/15657/commits/28579ed942b6839853dd9d4346be33c17cc80b3a